lib/toplevel.mli

```diff
@@ -19,8 +19,7 @@
 type t = {
   vpad : int;
   hpad : int;
-  line : int;
-  command : string list;
+  command : string list Location.loc;
```

Comment:
The location is computed in lexer_top, do you think it would be better to not update the location updates in the lexer?

Reply:
Having a location in the lexer make sense. Since the `loc_end` field is ignored anyway, I think it should be ignored earlier to avoid computing it in other places. Keeping it in the lexer is fine.

Reply:
I simplified pp_line_directive and its calls, it's difficult to do more as we need the locations from the read_lines function as well in other places.

Reply:
What I meant is that we should have only a `Lexing.position` in `Toplevel.t`. The `loc_end` field should never be computed.

Reply:
The whole location is computed in the lexer anyway so we won't gain anything by cutting this field, but I managed to remove a lot of unused parameters here and there that makes the code easier to read.

Reply:
This was useful to remove computing it in `read_lines`. I see you removed that so I have less strong opinions about it.
Note that the full location is still not used and still makes a little harder to create a `Toplevel.t`.

Reply:
done